File: packages/dart/lib/src/crs/utm_zone_for.dart
import 'utm_config.dart';

int utmZoneFor(double lng) {
  for (final z in UtmConfig.zones) {
    if (lng >= z.lngMin && lng < z.lngMax) return z.zone;
  }
  return ((lng + 180) / 6).floor() + 1;
}

File: packages/rust/src/crs/utm_zone_for.rs
use crate::data::crs::ZONES;

pub fn utm_zone_for(lng: f64) -> i32 {
    for z in ZONES {
        if lng >= z.lng_min && lng < z.lng_max {
            return z.zone;
        }
    }
    (((lng + 180.0) / 6.0).floor() as i32) + 1
}
